pkg/apis/admissionregistration/types.go

```diff
@@ -177,6 +187,17 @@ type Webhook struct {
 	// +optional
 	FailurePolicy *FailurePolicyType
 
+	// matchPolicy defines how the "rules" field is applied when a request is made
+	// to a different API group or version of a resource listed in "rules".
+	// Allowed values are "Exact" or "Equivalent".
+	// - Exact: match requests only if they exactly match a given rule. For example, if an object can be modified
+	// via API version v1 and v2, and "rules" only includes "v1", do not send a request to "v2" to the webhook.
+	// - Equivalent: match requests if they modify a resource listed in rules via another API group or version.
+	// For example, if an object can be modified via API version v1 and v2, and "rules" only includes "v1",
+	// a request to "v2" should be converted to "v1" and sent to the webhook.
```

Comment:
Without colocation in mind this doc really sounds cryptic. "Convertible" would maybe be a better term than the super generic "Equivalent".

If we stay with Equivalent, something like this is much more concrete:

match requests modifying resources listed in rules exactly, or those in different version in the same group, or those which are colocated in storage, i.e. represent the same objects.

Reply:
Agree. Expanding on "via another API group or version" should help, maybe like @sttts suggests.

I'm not sure if we should introduce the colocated term (which I'm guessing is unfamiliar to most readers and needs to be explained a bit?) or if should try to build on intuition, e.g. "if an object can be modified at either API group version, they are equivalent"...

Reply:
I think "equivalent" is better than "convertible". An apps/v1 Deployment object sent to an `apps/v1/deployments/.../status` *subresource* can be converted to an apps/v1beta1 Deployment, but would not be sent to a webhook registered for `apps/v1beta1/deployments` (not registered for the status subresource)

I'll update the doc to try to clarify this

Reply:
updated to:
```
// matchPolicy defines how the "rules" list is used to match incoming requests.
// Allowed values are "Exact" or "Equivalent".
//
// - Exact: match a request only if it exactly matches a specified rule.
// For example, if deployments can be modified via apps/v1, apps/v1beta1, and extensions/v1beta1,
// but "rules" only included `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]`,
// a request to apps/v1beta1 or extensions/v1beta1 would not be sent to the webhook.
//
// - Equivalent: match a request if modifies a resource listed in rules, even via another API group or version.
// For example, if deployments can be modified via apps/v1, apps/v1beta1, and extensions/v1beta1,
// and "rules" only included `apiGroups:["apps"], apiVersions:["v1"], resources: ["deployments"]`,
// a request to apps/v1beta1 or extensions/v1beta1 would be converted to apps/v1 and sent to the webhook.
//
// Defaults to "Exact"
// +optional
```